Modules/Filtering/ImageFeature/include/itkFastBilateralImageFilter.hxx

```diff
@@ -0,0 +1,311 @@
+  // used during down-sampling and reconstruction
+  InputPixelType intensityMin;
+
+  // Define the GridType
+  // These are pointers to the source and destination of the blurring filter
+  // Data from the input will be sorted into the first two images while the
+  // second two images will provide the output data.
+  // The parameters are determined by the size of the input image and the
+  // values of m_DomainSigma and m_RangeSigma
+  typename GridType::Pointer gridImage = GridType::New();
+  typename GridType::Pointer gridWeight;
+  typename GridType::Pointer gridImageOut;
+  typename GridType::Pointer gridWeightOut;
+
+  // The amount of padding around the grid images, required so that
+  // interpolation is not done outside of the grid during reconstruction
+  int padding = 2;
+
+  // Setup the higher dimensional grids (gridImage and gridWeight).
+  {
+    GridIndexType gridStartPos;
+    GridSizeType  gridSize;
+
+    // Convert domain sigmas from spacing units to pixel units
+    // for the blurring in the grid.
+    // The set method for m_DomainSigma uses spacing units to be
+    // consistent with the itkBilateralImageFilter implementation.
+    // When the data is placed into bins in the grid the
+    // itkDiscreteGaussianImageFilter will be run on the grid using
+    // imageSpacingOff.
+    InputImageSizeType inputSize = input->GetRequestedRegion().GetSize();
+    //  InputImageSizeType fullSize = input->GetLargestPossibleRegion().GetSize();
+
+    const InputImageSpacingType & spacing = input->GetSpacing();
+    for (size_t i = 0; i < ImageDimension; ++i)
+    {
+      domainSigmaInPixels[i] = m_DomainSigma[i] / spacing[i];
+      gridSize[i] = std::floor((inputSize[i] - 1) / domainSigmaInPixels[i]) + 1 + 2 * padding;
+    }
+
+    // Determine min/max intensities to calculate grid size in the intensity axis
+    using MinMaxCalculatorType = MinimumMaximumImageCalculator<InputImageType>;
+    typename MinMaxCalculatorType::Pointer calculator = MinMaxCalculatorType::New();
+
+    calculator->SetImage(input);
+    calculator->Compute();
+    intensityMin = calculator->GetMinimum();
+    InputPixelType intensityMax = calculator->GetMaximum();
+    InputPixelType intensityDelta = static_cast<InputPixelType>(intensityMax - intensityMin);
+    gridSize[ImageDimension] = static_cast<GridSizeValueType>((intensityDelta / m_RangeSigma) + 1 + 2 * padding);
+
+    for (size_t i = 0; i < ImageDimension + 1; ++i)
+    {
+      gridStartPos[i] = 0;
+    }
+
+    GridRegionType region;
+    region.SetSize(gridSize);
+    region.SetIndex(gridStartPos);
+    gridImage->SetRegions(region);
+  }
+
+  // Allocate the memory for pixel data
+  // We now have an empty grid. Two will be needed total,
+  // one will be a container for the down-sampled data and one
+  // will remember how many pixels were placed into each bin
+  gridImage->Allocate();
+  // Init values of image to 0
+  gridImage->FillBuffer(0.0);
```

Comment:
Nowadays, no need to call `Allocate(); FillBuffer(0.0)`, just call:

    gridImage->AllocateInitialized();